### PR TITLE
Add "Optimistic State" architecture cookbook recipe

### PR DESCRIPTION
**Pull request by miquelbeltran**
_Description of what this PR is changing or adding, and why:_

This is the first of several PRs that add the app-architecture cookbook recipes.

This one includes the recipe for "Optimistic State".

Some considerations:

- Code sample is under new folder `examples/app-architecture`, I plan to add the rest of cookbook recipe code there as well.
- Created a template for a recipe index under `src/content/app-architecture/cookbook/index.md` this is in the same sub-path as the App-architecture guidelines in #11300.
- I did not add these recipes to the general "cookbook" of the site. But maybe it should? Let me know if we should put them there.
- I did not create this PR on top of #11300 as it didn't seem necessary, since it has no direct dependencies.
- Page is NOT accessible from the sidebar, instead you have to navigate to `/app-architecture/cookbook/optimistic-state/`: https://flutter-docs-prod--pr11394-mb-optimistic-state-r6prt6uc.web.app/app-architecture/cookbook/optimistic-state

cc. @ericwindmill 

_Issues fixed by this PR (if any):_

- Part of #11374

_PRs or commits this PR depends on (if any):_

- Depends on #11300

## Presubmit checklist

- [ ] This PR is marked as draft with an explanation if not meant to land until a future stable release.
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/content/app-architecture/cookbook/optimistic-state.md [hunk 1]
```diff
@@ -0,0 +1,600 @@
...
+    // Simulate a network request
+    await Future.delayed(const Duration(seconds: 1));
+    // Fail after one second
+    throw Exception('Failed to subscribe');
+  }
+}
+```
+
+[Flutter Architecture guidelines]:/app-architecture
+[Command pattern]:/app-architecture/cookbook/command
```
Comment (miquelbeltran) [inline src/content/app-architecture/cookbook/optimistic-state.md:600]: FYI, these locations are still not available, so the urls will 404 atm

**Comment by ericwindmill** [comment]
> * I did not add these recipes to the general "cookbook" of the site. But maybe it should? Let me know if we should put them there.
> * Page is NOT accessible from the sidebar, instead you have to navigate to `/app-architecture/cookbook/optimistic-state/`: https://flutter-docs-prod--pr11394-mb-optimistic-state-r6prt6uc.web.app/app-architecture/cookbook/optimistic-state

@Sfshaza and @parlough -- I'm curious if you all have different opinions about the general info architecture of the architecture work Miguel and I have done.

I agree with both of these decisions. In a future PR, after #11300 lands, I want to expand and then add the index page for these architecture recipes to the side nav. For context, the recipes that Miguel has written for app architecture refer to the app architecture docs to varying degrees. For that reason, I believe they don't belong on the main Cookbook index page.

**Comment by sfshaza2** [comment]
@ericwindmill, I'd like @parlough's opinion on this, but I don't want a plethora of cookbooks. I'd like to incorporate these new recipes into a new category of the existing cookbook. When Parker re-implemented the website, he added the ability to add items to the sidenav a) more than once and b) regardless of where the files sit on the website. This is why we surfaced cookbook recipes in other parts of the sidenav, but we still keep them together on disk. I'd like to follow this process for these recipes.

**Comment by ericwindmill** [comment]
> @ericwindmill, I'd like @parlough's opinion on this, but I don't want a plethora of cookbooks. I'd like to incorporate these new recipes into a new category of the existing cookbook. When Parker re-implemented the website, he added the ability to add items to the sidenav a) more than once and b) regardless of where the files sit on the website. This is why we surfaced cookbook recipes in other parts of the sidenav, but we still keep them together on disk. I'd like to follow this process for these recipes.

I am 100% okay with these living in a new subdirectory under cookbook like /content/cookbook/architecture/. As you know, I love organization above all else  :)   I don't actually want these in the sidenav, because I don't want to over-complicate the sidenav of the site in general, regardless of this PR.  What I actually want in the side nav is just a page under 'App Architecture' that introduces and links to the architecture recipes that Miguel has written. (There are 6 or 7 total, btw).

@miquelbeltran can you make the following changes: 
* move the recipes to the main cookbook in a new subdir  
* move the index file you created up a level to /app-architecture and rename it 'Design Patterns'
* add the Design Patterns page to the side nav under the 'Recommendations' page?

In the near future when more cookbooks have landed, I want to update the design patterns page to be visually interesting, with images or a table or something.  I'll document that work in the umbrella ticket.

**Comment by miquelbeltran** [comment]
On it! I will move this and the ones in #11410 to the `/content/cookbook/architecture/` folder, I will also move the examples into `/examples/cookbook/architecture/`

**Comment by miquelbeltran** [comment]
image

**Comment by miquelbeltran** [comment]
I will have to remove the "Command pattern" link otherwise the site check fails, I will add this link once I create the PR for the Command.

**Comment by ericwindmill** [comment]
After CI passes, I'm going to merge this. Speak now if you disagree!

**Review by ericwindmill — APPROVED** [review]

**Pull request merged**